app/views/ladder/LadderView.coffee

```diff
@@ -71,19 +69,68 @@ module.exports = class LadderView extends RootView
 
     @displayTabContent = 'display: block'
 
+    @calcTimeOffset()
+    @mandate = @supermodel.loadModel(new Mandate()).model
+
     @loadLeague()
     @urls = require('core/urls')
 
-  checkForTournamentEnd: =>
-    return if @destroyed
-    return false if me.isAdmin()
-    $.get '/db/mandate', (data) =>
-      return if @destroyed
-      if data?[0]?.currentTournament isnt 'magic-rush'
-        @tournamentEnd = true
-        @displayTabContent = 'display: none'
+    if features.china
+      setTimeout =>
+        @checkTournamentEndInterval = setInterval @checkTournamentEnd(), 10000
```

Comment:
This is great but I don't think this will work in this form. The `@checkTournamentEnd()` will be evaluated and you'll instead pass the returned value into `setInterval` which won't check the tournament end.

Here's an example case:
`setInterval(console.log('hi'), 10000)` doesn't work because it evaluates and passes `undefined` into `setInterval`.
Instead I think what you want is `setInterval((() => console.log('hi')), 1000)` where the function is lazy until executed by `setInterval`.

Thus I recommend refactoring this to look like so:
```suggestion
        @checkTournamentEndInterval = setInterval((() => @checkTournamentEnd()), 10000)
```

Reply:
Or you can use `@checkTournamentEnd.bind(@)` instead of the arrow function. Both create a lazy executable function that setInterval can call.

Reply:
In fact, I return the function itself, so setInterval  can accept @checkTournamentEnd()  as parameter. The effect is the function will execute immediately then start the interval check

![image](https://user-images.githubusercontent.com/11880337/72029511-1f188900-32c1-11ea-9994-7f1e38a1c54d.png)

Reply:
Thanks for pointing this out, I totally missed that! Although this is really clever, I think this pattern might be less readable and make understanding the setInterval more difficult.

Is there a reason for doing it this way rather than using `.bind(@)`?

Reply:
just in order to execute for the first time to avoid long waiting time for `setInterval`
as we set the time interval to 2000ms, no need for that any more